### PR TITLE
Switch to ophyd async

### PR DESCRIPTION
**Pull request by rosesyrett**
Simple switch to replace ophyd.v2 references with ophyd_async

### CODE REVIEW

**Review by DominicOram — APPROVED** [review]
Looks good to me, @Tom-Willemsen I think this is mergable, right?

**Comment by Tom-Willemsen** [comment]
So I think now that we have https://github.com/bluesky/ophyd-async/pull/19 in ophyd-async, we need to remove the array-reshaping bits from within the pin image recognition device - we were previously using a fork of ophyd that supported `ndarray` but *didn't* do the reshaping at the ophyd level.

I started doing this in a draft PR at https://github.com/DiamondLightSource/dodal/pull/162 but still need to look at why one of the unit tests is failing there (I haven't had time for this recently). I don't think it will work if the reshaping is effectively being done both here and in ophyd.

File: pyproject.toml [hunk 1]
```diff
@@ -14,17 +14,18 @@ classifiers = [
...
     "pillow",
     "requests",
     "graypy",
     "pydantic<2.0",
-    "opencv-python-headless",                                                                  # For pin-tip detection.
-    "aioca",                                                                                   # Required for CA support with Ophyd V2.
-    "p4p",                                                                                     # Required for PVA support with Ophyd V2.
+    "opencv-python-headless", # For pin-tip detection.
+    "aioca",                  # Required for CA support with Ophyd V2.
+    "p4p",                    # Required for PVA support with Ophyd V2.
```
Comment (Tom-Willemsen) [inline pyproject.toml:28]: nit-pick: maybe we could replace the reference in these comments from "Ophyd V2" to "Ophyd async" now that the project has been renamed?

```suggestion
    "aioca",                  # Required for CA support with ophyd-async.
    "p4p",                    # Required for PVA support with ophyd-async.
```

**Comment by rosesyrett** [comment]
Ok; no problem. This PR can wait until that is done, or we can close it in favour of #162 since that seems to also remove references to `ophyd.v2`.

**Comment by Tom-Willemsen** [comment]
I'll try to get the other one in a state ready to merge as soon as I have a spare 5 mins and then close off this one once that is merged. Keith/Calum were also asking about this indirectly this morning...

**Pull request closed without merging**